sql-cli/sql_cli/workflow_directory_parser.py

```diff
@@ -0,0 +1,245 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+from pathlib import Path
+from typing import Iterable
+
+import airflow
+import frontmatter
+import yaml
+from airflow.models import BaseOperator
+from yaml.scanner import ScannerError
+
+from astro.sql import LoadFileOperator
+from astro.sql.operators.transform import TransformOperator
+from sql_cli.constants import (
+    GENERATED_WORKFLOW_INCLUDE_DIRECTORY,
+    LOAD_FILE_OPERATOR,
+    SQL_FILE_TYPE,
+    YAML_FILE_TYPE,
+)
+from sql_cli.operators.load_file import build_load_file_operator_content, get_load_file_instance
+from sql_cli.utils.jinja import find_template_variables
+
+
+class WorkflowFile:
+    """
+    A Workflow is equivalent to a chain of corresponding operators in Astro SDK.
+
+    :param root_directory: The root directory path of the project.
+    :param path: The path to the workflow file.
+    :param file_type: The type of the workflow file. e.g. sql or yaml
+    :param target_directory: The target directory path for the executable workflow file.
+    """
+
+    def __init__(self, root_directory: Path, path: Path, target_directory: Path, file_type: str) -> None:
+        self.root_directory = root_directory
+        self.path = path
+        self.target_directory = target_directory
+        self.raw_content = self.path.read_text()
+        self.file_type = file_type
```

Comment:
`WorkflowFile` should not have the file_type if it does not use it. We instantiate different WorkflowFile objects based on the file type already.